flexmeasures/data/schemas/sensors.py

```diff
@@ -239,36 +273,58 @@ def _deserialize(
             sensor = SensorIdField(unit=self.to_unit)._deserialize(
                 value["sensor"], None, None
             )
-
             return sensor
 
+        elif isinstance(value, list):
+            if self.return_magnitude is True:
+                current_app.logger.warning(
+                    "Deserialized time series will include Quantity objects in the future. Set `return_magnitude=False` to trigger the new behaviour."
+                )
+            field = fields.List(
+                fields.Nested(
+                    TimedEventSchema(
+                        timezone=self.timezone,
+                        value_validator=self.value_validator,
+                        to_unit=self.to_unit,
+                        default_src_unit=self.default_src_unit,
+                        return_magnitude=self.return_magnitude,
+                    )
+                )
+            )
+            return field._deserialize(value, None, None)
+
         elif isinstance(value, str):
             try:
                 return ur.Quantity(value).to(self.to_unit)
             except DimensionalityError as e:
                 raise FMValidationError(
                     f"Cannot convert value `{value}` to '{self.to_unit}'"
                 ) from e
-        else:
-            if self.default_src_unit is not None:
-                return self._deserialize(
-                    f"{value} {self.default_src_unit}", attr, obj, **kwargs
-                )
 
+        elif self.default_src_unit is not None:
```

Comment:
I don't entirely follow. We would only hit the `else` statement if `self.default_src_unit is None`, so e.g. when the user did not use the `soc-unit` field.

But okay, if I assume that is what you meant, then my argument would be that we want the error message to nudge users into specifying a string quantity, rather than having them keep using a numeric value while reminding them to specify the unit via a separate field.

I've basically started to hide the existence of the `soc-unit` field.

Reply:
Well, I think it's a bit better if we check the type explicitly as python is not taking care that we only get the types that are listed in the argument type annotation. The else should raise a TypeError to indicate that a wrong type has been provided.